test/unit/org/apache/cassandra/index/sai/cql/NativeIndexDDLTest.java

```diff
@@ -264,16 +263,86 @@ public void shouldNotFailCreateWithTupleType()
     }
 
     @Test
-    public void shouldFailCreateWithInvalidCharactersInColumnName()
+    public void testSimpleQuotedIdentifierColumn()
     {
-        String invalidColumn = "/invalid";
-        createTable(String.format("CREATE TABLE %%s (id text PRIMARY KEY, \"%s\" text)", invalidColumn));
+        createTable("CREATE TABLE %s (\"key\" int PRIMARY KEY, \"value\" text)");
+        createIndex("CREATE CUSTOM INDEX ON %s(\"value\") USING 'StorageAttachedIndex'");
 
-        assertThatThrownBy(() -> executeNet(String.format("CREATE CUSTOM INDEX ON %%s(\"%s\")" +
-                                                          " USING 'StorageAttachedIndex'", invalidColumn)))
-                .isInstanceOf(InvalidQueryException.class)
-                .hasMessage(String.format("Column '%s' is longer than the permissible name length of %d characters or" +
-                                          " contains non-alphanumeric-underscore characters", invalidColumn, SchemaConstants.NAME_LENGTH));
+        execute("INSERT INTO %s (\"key\", \"value\") VALUES (1, 'value1')");
+        execute("INSERT INTO %s (\"key\", \"value\") VALUES (2, 'value2')");
+
+        assertRows(execute("SELECT * FROM %s WHERE \"value\" = 'value1'"), row(1, "value1"));
+        assertRows(execute("SELECT * FROM %s WHERE \"value\" = 'value2'"), row(2, "value2"));
+    }
+
+    private String intoColumnDefs(String[] columnNames)
+    {
+        StringBuilder sb = new StringBuilder();
+        for (String columnName : columnNames)
+        {
+            sb.append(columnName).append(" int,");
+        }
+        return sb.toString();
+    }
+
+    @Test
+    public void testQuotedIdentifierWithSpecialCharsAndLong()
+    {
+        String[] columnNames
+        = new String[]{ "\"user name\"",
+                        "\"userCountry\"",
+                        "\"user-age\"",
+                        "\"/user/age\"",
+                        "\"userage\"",
+                        "\"a very very very very very very very very long field\"",
+                        "\"   a_very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very(very)very_very "
+                        + "_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_very \""
+        };
+
+        createTable("CREATE TABLE %s (key int,"
+                    + intoColumnDefs(columnNames)
+                    + "PRIMARY KEY (key))");
+        for (String columnName : columnNames)
+            execute("CREATE CUSTOM INDEX ON %s (" + columnName + ") USING 'StorageAttachedIndex'");
```

Comment:
I was wondering why this uses `CQLTester#execute` rather than the usual `CQLTester#createIndex`. It turns out that `createIndex` will fail with a timeout because it uses `CQLTester.getCreateIndexName`. 

That method hasn't been updated to work accordingly to the modified `IndexMetadata.generateDefaultIndexName`. Thus, `CQLTester#createIndex` returns an index name different to the actual index name. As a result, it keeps waiting for the wrong index build until it times out.

Reply:
Great catch. The reason for the failure is not the change to `generateDefaultIndexName`, but removing the restriction for qualified column names in SAI. The original behavior of `generateDefaultIndexName` to remove non-word chars produce different name.
I will look to fix it.

Reply:
I investigated and I see that it was my change ot `generateDefaultIndexNam`, which broke `createIndex`. I was incorrect with my understanding above. Fixing.

Reply:
Actually matcher doesn't find the column name if it's qualified.

Reply:
Fixed in 168e314f1c7e198027a3fe30eb86ab63cd06ffc1

Reply:
This was not enough, so I reverted from `createIndex` to `execute`, since as CQL tester assumes no qualified names it always lower case the name. I will look if I can fix it.

Reply:
Finally I fixed `createIndex` to work with majority cases with qualified column name.